e2e/playwright/st_headings_divider_test.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from playwright.sync_api import Page, expect
+
+from conftest import ImageCompareFunction
+
+
+def test_header_display(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test that st.header renders correctly with dividers."""
+    header_elements = app.locator(".stHeadingContainer")
+    expect(header_elements).to_have_count(16)
+
+    for i, element in enumerate(header_elements.all()):
```

Comment:
nit: I used this enumeration in most of the playwright tests. But today, William suggested on one of my PRs to actually list all screenshots with more descriptive names, e.g.:

<img width="765" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/337685c7-c2bd-4a39-b1ca-293cd9bf5ca1">

I'm not yet completely sold on doing that in all the tests, so feel free to leave it as is... but I do see small advantages with this.